python/code_challenges/linked_list_zip.py

```diff
@@ -1,2 +1,21 @@
 def zip_lists(a, b):
-    pass
+    a_node = a.head
+    b_node = b.head
+    # account for empty LL in arguments
+    if a.head is None:
+        return b
+    if b.head is None:
+        return a
+
+    while a_node and b_node:
+        # store next pointers
+        a_next = a_node.next
+        b_next = b_node.next
+        # swap
+        b_node.next = a_next
+        a_node.next = b_node
+        # drive
+        a_node = a_next
+        b_node = b_next
```

Comment:
Looks like the `test_a_shorter` test is failing because you aren't maintaining your reference to b.next upon looping.

If list a is shorter, your function just ends, as it's currently written.